sycl/test-e2e/README.md

```diff
@@ -448,54 +448,23 @@ system.
-  * The vast majority of `RUN:` lines that do not execute the test binaries are
-  needed to either set up files prior to compilation, or to compile the binary,
-  as such `RUN:` lines are by default considered as build lines. In the case
-  that we need to run a line on the `run-only` system, and it does not make
-  sense to mark them with `%{run}` or `%{run-unfiltered-devices}`, we can mark a
-  line with `%if run-mode` to specifically make the line a run line. This
-  situation usually appears when we need to run a command in response to the
-  execution of the test binary.
-
-* Currently the `build-only` mode does not support logic to properly assess the
-features in `REQUIRES`/`UNSUPPORTED` to know if a test can be built in the
-system environment, or for `spir64`. Only tests that are marked with `REQUIRES:
-build-and-run-mode` or `UNSUPPORTED: true` are skipped. Thus if a test will fail
-building for the build environment we have on CI or for `spir64` we will need to
-mark this as `REQUIRES: build-and-run-mode`. This is only temporary solution,
-until further work is done to properly mark tests as unsupported on `build-only`
-based on features.
-
-* CPU and FPGA AOT tests are currently expected to fail when compiling and
-executing on separate machines. These failures occur on the `run-only` side,
-because during compilation the host machine's CPU architecture is targeted,
-which may be different than that of the running machine. These tests are marked
-as `REQUIRES: build-and-run-mode` as a result, until they can be refactored to
-compile for the architectures that will be used on the run side.
-
-#### Falling back to `full` testing mode on `run-only`
-
-To not lose coverage of tests marked as `REQUIRES: build-and-run-mode` when
-using `run-only` mode, lit can be called using
-`--param fallback-to-build-if-requires-build-and-run=True`. When this option is
-enabled in `run-only` mode, tests marked as requiring `build-and-run-mode` will
-fallback to running on `full` mode, instead of being reported as unsupported.
+  with `XFAIL` on a device specific feature. If it is expected to fail for any
+  device at run-time we can use `XFAIL: run-mode`, this is because otherwise the
+  test would compile and pass on `build-only` mode and be reported as an `XPASS`.
+
+* To separate compilation and execution of tests, `RUN:` lines are filtered in
+`build-only` and `run-only` mode based on the presence of "run" expansions.
+  * Any line that is meant to execute the test binary should be marked with
+  `%{run}` or `%{run-unfiltered-devices}` so that it is not ran in `build-only`.
+  * The `%{run-aux}` expansion can be used if a `RUN:` line that does not
+  execute a test binary needs to be ran in `run-only`.
+
+* Compiling for CPU AOT will target the ISA of the host CPU, thus compiling
+these tests on a different system will lead to failures if the build system and
+run system support different ISAs.
```

Comment:
To accommodate that, such compilations are delayed to "run" stage by using `%{run-aux}` markup.